src/ngraph/op/scatter_nd_update.cpp

```diff
@@ -0,0 +1,95 @@
+//*****************************************************************************
+
+#include "ngraph/op/scatter_nd_update.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/shape.hpp"
+using namespace std;
+using namespace ngraph;
+
+static int INPUTS = 0;
+static int INDICES = 1;
+static int UPDATES = 2;
+
+constexpr NodeTypeInfo op::v3::ScatterNDUpdate::type_info;
+
+shared_ptr<Node> op::v3::ScatterNDUpdate::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<ScatterNDUpdate>(
+        new_args.at(INPUTS), new_args.at(INDICES), new_args.at(UPDATES));
+}
+
+void op::v3::ScatterNDUpdate::validate_and_infer_types()
+{
+    element::Type inputs_et = get_input_element_type(INPUTS);
+    element::Type indices_et = get_input_element_type(INDICES);
+    element::Type updates_et = get_input_element_type(UPDATES);
+
+    const PartialShape& inputs_shape = get_input_partial_shape(INPUTS);
+    const PartialShape& indices_shape = get_input_partial_shape(INDICES);
+    const PartialShape& updates_shape = get_input_partial_shape(UPDATES);
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_et == element::i32 || indices_et == element::i64,
+                          "Indices element type must be i64 or i32");
+
+    NODE_VALIDATION_CHECK(
+        this, updates_et == inputs_et, "Updates element type must be the same as inputs");
+
+    NODE_VALIDATION_CHECK(this,
+                          indices_shape.rank().is_dynamic() ||
+                              indices_shape.rank().get_length() >= 1,
+                          "Indices rank is expected to be at least 1");
+
+    NODE_VALIDATION_CHECK(this,
+                          inputs_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
+                              indices_shape[indices_shape.rank().get_length() - 1].get_length() <=
+                                  inputs_shape.rank().get_length(),
+                          "Last dimension of indices can be at most the rank of inputs");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        inputs_shape.rank().is_dynamic() || indices_shape.rank().is_dynamic() ||
+            updates_shape.rank().is_dynamic() ||
+            updates_shape.rank().get_length() ==
+                indices_shape.rank().get_length() + inputs_shape.rank().get_length() -
+                    indices_shape[indices_shape.rank().get_length() - 1].get_length() - 1,
+        "Rank of updates must be rank of inputs + rank of indices - last dimension of indices - 1");
+
+    bool compatible = true;
+    if (inputs_shape.is_static() && indices_shape.is_static() && updates_shape.is_static())
+    {
+        size_t indices_rank = indices_shape.rank().get_length();
+        size_t updates_rank = updates_shape.rank().get_length();
+        for (size_t i = 0; i < indices_rank - 1; i++)
+        {
+            compatible = compatible && updates_shape[i].same_scheme(indices_shape[i]);
```

Comment:
Do the NODE_VALIDATION_CHECK here so you can say what was invalid as soon as you find it.